source/event_loop.c

```diff
@@ -462,10 +566,11 @@ int aws_event_loop_subscribe_to_io_events(
     aws_event_loop_on_event_fn *on_event,
     void *user_data) {
 
-    AWS_ASSERT(event_loop->vtable && event_loop->vtable->subscribe_to_io_events);
-    return event_loop->vtable->subscribe_to_io_events(event_loop, handle, events, on_event, user_data);
+    if (event_loop->vtable && event_loop->vtable->subscribe_to_io_events) {
```

Comment:
There's some inconsistency here that feels off.  Most calls we assert the function pointer and invoke it.  But here we have two functions that check it and return an error if it's not set.  Is there something sensible and consistent we could do?